examples/release_highlights/plot_release_highlights_0_23_0.py

```diff
@@ -0,0 +1,140 @@
+# now support :ref:`sample weights <sw_hgbdt>`, and a new Poisson loss function
+# was implemented. Also, an automatic early-stopping criterion was added:
+# early-stopping is enabled by default when the number of samples exceeds 10k.
+# The following snippet illustrates the use of the Poisson loss.
+
+from sklearn.experimental import enable_hist_gradient_boosting  # noqa
+from sklearn.ensemble import HistGradientBoostingRegressor
+
+X = rng.randn(n_samples, n_features)
+y = ((10 + X[:, 5]) * 10).astype(int)  # easy target correlated with X[:, 5]
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=rng)
+reg = HistGradientBoostingRegressor(loss='poisson', learning_rate=.01)
+reg.fit(X_train, y_train)
+print(reg.score(X_test, y_test))
+
+##############################################################################
+# Finally, users can now define :ref:`monotonic constraints
+# <monotonic_cst_gbdt>` to constrain the predictions based on the variations of
+# specific features. In the following example, we construct a target that is
+# generally positively correlated with the first feature, with some noise.
+# Applying monotoinc constraints allows the prediction to capture the global
+# effect of the first feature, instead of fitting the noise.
+from sklearn.inspection import plot_partial_dependence
+from matplotlib import pyplot as plt
+
+n_samples = 5000
+X = rng.randn(n_samples, 2)
+noise = rng.normal(loc=0.0, scale=0.01, size=n_samples)
+y = (5 * X[:, 0] + np.sin(10 * np.pi * X[:, 0]) - noise)
+
+gbdt_no_cst = HistGradientBoostingRegressor().fit(X, y)
+gbdt_cst = HistGradientBoostingRegressor(monotonic_cst=[1, 0]).fit(X, y)
+
+disp = plot_partial_dependence(
+    gbdt_no_cst, X, features=[0], feature_names=['feature 0'],
+    line_kw={'linewidth': 4, 'label': 'unconstrained'})
+plot_partial_dependence(gbdt_cst, X, features=[0],
+    line_kw={'linewidth': 4, 'label': 'constrained'}, ax=disp.axes_)
+disp.axes_[0, 0].plot(X[:, 0], y, 'o', alpha=.3, zorder=-1, label='samples')
+disp.axes_[0, 0].set_ylim(-3, 3); disp.axes_[0, 0].set_xlim(-1, 1)
+plt.legend(); plt.show()
+
+##############################################################################
+# Sample-weight support for Lasso and ElasticNet
+# ----------------------------------------------
+# The two linear regressors :class:`~sklearn.linear_model.Lasso` and
+# :class:`~sklearn.linear_model.ElasticNet` now support sample weights:
+
+from sklearn.datasets import make_regression
+from sklearn.linear_model import Lasso
+
+X, y = make_regression(n_samples, n_features, random_state=rng)
+sample_weight = rng.rand(n_samples)
+X_train, X_test, y_train, y_test, sw_train, sw_test = train_test_split(
+    X, y, sample_weight, random_state=rng)
+reg = Lasso()
+reg.fit(X_train, y_train, sample_weight=sw_train)
+print(reg.score(X_test, y_test, sw_test))
+
+##############################################################################
+# Usage of the Boston dataset
```

Comment:
There's room for improvement @jnothman @adrinjalali , I'm happy to consider your suggestions

(also feel free to push directly)

Reply:
Yeah, maybe we should drop this from here.